File: erpyemen/customizations/purchase_invoice.py
import frappe

def update_item_expiry(doc, method):
    for row in doc.items:
        if row.expiry_date:
            frappe.db.set_value("Item", row.item_code, "end_of_life", row.expiry_date)

def update_item_prices(doc, method):
    for item in doc.items:
        if not item.item_code or not item.sales_price:
            continue

        price_list = "البيع القياسية"  # Change if you use a different price list

        existing_price = frappe.get_all("Item Price", filters={
            "item_code": item.item_code,
            "price_list": price_list,
            "selling": 1
        }, fields=["name"])

        if existing_price:
            # Update existing price
            ip_doc = frappe.get_doc("Item Price", existing_price[0].name)
            ip_doc.price_list_rate = item.sales_price
            ip_doc.save()
        else:
            # Create new selling price
            frappe.get_doc({
                "doctype": "Item Price",
                "item_code": item.item_code,
                "price_list": price_list,
                "price_list_rate": item.sales_price,
                "selling": 1
            }).insert()            

File: erpyemen/customizations/purchase_invoice.js
frappe.ui.form.on("Purchase Invoice", {
    onload: function(frm) {
        frm.fields_dict['items'].grid.get_field('uom').get_query = function(doc, cdt, cdn) {
            let child = locals[cdt][cdn];

            if (!child.item_code) return;

            return {
                query: "erpyemen.customizations.sales_invoice.get_items_uoms",
                filters: {
                    parent: child.item_code
                }
            };
        };
    },
    //update price list rate 
    after_save: function(frm) {
        frm.doc.items.forEach(item => {
            if (!item.item_code || !item.rate) return;
            frappe.db.get_doc('Item', item.item_code).then(item_doc => {
                let rate_in_stock_uom = item.rate;

                if (item.uom !== item_doc.stock_uom && item.conversion_factor) {
                    rate_in_stock_uom = item.rate / item.conversion_factor;
                }

                // Now update only existing Item Price
                frappe.call({
                    method: "frappe.client.get_list",
                    args: {
                        doctype: "Item Price",
                        filters: {
                            item_code: item.item_code,
                            price_list: "شراء القياسية",
                            buying: 1
                        },
                        fields: ["name"]
                    },
                    callback: function(res) {
                        if (res.message && res.message.length > 0) {
                            let item_price_name = res.message[0].name;

                            frappe.call({
                                method: "frappe.client.set_value",
                                args: {
                                    doctype: "Item Price",
                                    name: item_price_name,
                                    fieldname: {
                                        price_list_rate: rate_in_stock_uom
                                    }
                                }
                            });
                        }
                    }
                });
            });
        });
    },
  

});
